routes/api/v1/dashboards.js

```diff
@@ -0,0 +1,69 @@
+const Dashboard = require('../../../models/dashboards.js')
+const { Widget } = require('../../../models/widgets.js')
+
+const express = require('express')
+const router = express.Router()
+
+router.get('/', (request, response, next) => {
+  Dashboard.find().exec()
+    .then( dashboard => response.json( dashboard ) )
+    .catch( error => next( error ) )
+})
+
+router.post('/', (request, response, next) => {
+  Dashboard.create( request.body )
+    .then( dashboard => response.json( dashboard ) )
+    .catch( error => next( error ) )
+})
+
+router.get('/:id', (request, response, next) => {
+  Dashboard.findById( request.params.id ).exec()
+    .then( dashboard => response.json( dashboard ) )
+    .catch( error => next( error ) )
+})
+
+router.put('/:id', (request, response, next) => {
+  Dashboard.findByIdAndUpdate( request.params.id, request.body ).exec()
+    .then( dashboard => response.json( dashboard ) )
+    .catch( error => next( error ) )
+})
+
+router.delete('/:id',(request, response, next) => {
+  Dashboard.findByIdAndRemove( request.params.id ).exec()
+    .then( dashboard => response.json( dashboard ) )
+    .catch( error => next( error ) )
+})
+
+// get all widgets by dashboard_id
+// GET dashboards/:id/widgets/
+
+// router.get('/:id/widgets/', (request, response, next) => {
+//   const findAllWidgets = widgetData => dashboard => {
+//     return dashboard.widgets.find().exec()
+//   }
+//   Widget.find({}, 'widgets'
+//   Dashboard.findById( request.params.id ).exec()
+//     .then( ).exec())
+//     .then( dashboard => response.json( dashboard ) )
+//     .catch( error => next( error ) )
+// })
+
+router.post('/:id/widgets', (request, response, next) => {
+  const addWidget = widgetData => dashboard => {
```

Comment:
Please define this function in its own module - I propose a `dashboard` folder under routes that may contain functions like this...